packages/util-dynamodb/src/convertToAttr.ts

```diff
@@ -22,44 +22,52 @@ export const convertToAttr = (data: NativeAttributeValue, options?: marshallOpti
 };
 
 const convertToListAttr = (data: NativeAttributeValue[], options?: marshallOptions): { L: AttributeValue[] } => ({
-  L: data.map((item) => convertToAttr(item, options)),
+  L: data
+    .filter((item) => !options?.removeUndefinedValues || (options?.removeUndefinedValues && item !== undefined))
```

Comment:
Can the undefined check be moved to `convertToAttr`? Instead of under each convert functions.

Reply:
I tried to attempt it in first commit [`3578da6` (#1840)](https://github.com/aws/aws-sdk-js-v3/pull/1840/commits/3578da69049dd5cf8357ca9197a59ff394f58a4d)

The issue is that `convertToAttr` is exported and should throw if `undefined` is passed to it.
Let me re-attempt it to reduce duplication, and post-process undefined value returned by convertToAttr as PR is now in a good shape.

Reply:
The issue is `convertToAttr` is expected to return `AttributeValue`.
If it receives undefined and `removeUndefinedValues` is not set, it should throw. That's why pre-processing was added in map/set/list.

I can change `convertToAttr` return value to `AttributeValue | undefined` and perform post-processing in map/reduce operations to remove undefined values being returned.

The latter is better, as removeUndefinedValues check would be limited to `convertToScalarAttr` function.

Reply:
Update: Returning `AttributeValue | undefined` from `convertToAttr` makes the implementation more complex as `convertToAttr` is exported.

Experimental [diff.patch.txt](https://github.com/aws/aws-sdk-js-v3/files/5737466/diff.patch.txt) attached.

Reply:
I see. Thank you a lot for putting this up!